### PR TITLE
reverse-string: add approach introduction

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: exercises/practice/reverse-string/.approaches/performance/content.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Performance
+
+On the [approaches page][approaches], we described several approaches:
+
+1. Using LINQ
+2. Using `Array.Reverse()`
+3. Using a `StringBuilder`
+
+In this document, we'll find out which of these approaches is the most performant one.
```
Comment (iHiD) [inline exercises/practice/reverse-string/.approaches/performance/content.md:9]: I don't love "document" here but I don't have a better suggestion.

Comment (ErikSchierboom) [reply exercises/practice/reverse-string/.approaches/performance/content.md]: I've restructured things a bit

Comment (ErikSchierboom) [reply exercises/practice/reverse-string/.approaches/performance/content.md]: I think `Span<T>` is really its own approach, and the `performance` document should link to it.

File: exercises/practice/reverse-string/.approaches/array-reverse/content.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return new string(chars);
+    }
+}
+```
+
+The `string` class' [`ToCharArray()`][to-char-array] method returns the string's characters as a `char[]`.
+
+```exercism/caution
+The `char[]` returned by `ToCharArray()` is a **copy** of the `string`'s characters.
+Modifying the values in the `char[]` does not update the `string` it was created from.
```
Comment (ErikSchierboom) [inline exercises/practice/reverse-string/.approaches/array-reverse/content.md:21]:
```suggestion
Modifying the values in the `char[]` does **not** update the `string` it was created from.
```

**Review by iHiD — APPROVED** [review]

**Pull request merged**